mecanum_drive_controller/src/mecanum_drive_controller.cpp

```diff
@@ -0,0 +1,506 @@
+  auto current_ref = *(input_ref_.readFromRT());
+  const auto age_of_last_command = time - current_ref->header.stamp;
+  if (age_of_last_command <= ref_timeout_ || ref_timeout_ == rclcpp::Duration::from_seconds(0))
+  {
+    tf2::Quaternion quaternion;
+    quaternion.setRPY(0.0, 0.0, params_.kinematics.base_frame_offset.theta);
+    /// \note The variables meaning:
+    /// rotation_from_base_to_center: Rotation transformation matrix, to transform from base frame to center frame
+    /// linear_trans_from_base_to_center: offset/linear transformation matrix, to transform from base frame to center frame
+
+    tf2::Matrix3x3 rotation_from_base_to_center = tf2::Matrix3x3((quaternion));
+    tf2::Vector3 velocity_in_base_frame_w_r_t_center_frame_ =
+      rotation_from_base_to_center *
+      tf2::Vector3(reference_interfaces_[0], reference_interfaces_[1], 0.0);
+    tf2::Vector3 linear_trans_from_base_to_center = tf2::Vector3(
+      params_.kinematics.base_frame_offset.x, params_.kinematics.base_frame_offset.y, 0.0);
+
+    velocity_in_center_frame_linear_x =
+      velocity_in_base_frame_w_r_t_center_frame_.x() +
+      linear_trans_from_base_to_center.y() * reference_interfaces_[2];
+    velocity_in_center_frame_linear_y =
+      velocity_in_base_frame_w_r_t_center_frame_.y() -
+      linear_trans_from_base_to_center.x() * reference_interfaces_[2];
+    velocity_in_center_frame_angular_z = reference_interfaces_[2];
+
+    double w_front_left_vel =
+      1.0 / params_.kinematics.wheels_radius *
+      (velocity_in_center_frame_linear_x - velocity_in_center_frame_linear_y -
+       params_.kinematics.wheels_k * velocity_in_center_frame_angular_z);
+    double w_back_left_vel =
+      1.0 / params_.kinematics.wheels_radius *
+      (velocity_in_center_frame_linear_x + velocity_in_center_frame_linear_y -
+       params_.kinematics.wheels_k * velocity_in_center_frame_angular_z);
+    double w_back_right_vel =
+      1.0 / params_.kinematics.wheels_radius *
+      (velocity_in_center_frame_linear_x - velocity_in_center_frame_linear_y +
+       params_.kinematics.wheels_k * velocity_in_center_frame_angular_z);
+    double w_front_right_vel =
+      1.0 / params_.kinematics.wheels_radius *
+      (velocity_in_center_frame_linear_x + velocity_in_center_frame_linear_y +
+       params_.kinematics.wheels_k * velocity_in_center_frame_angular_z);
+
+    // Set wheels velocities:
+
+    command_interfaces_[0].set_value(w_front_left_vel);
+    fprintf(stderr, " command_interfaces_[0] = %f \n", w_front_left_vel);
+    command_interfaces_[1].set_value(w_back_left_vel);
+    fprintf(stderr, " command_interfaces_[1] = %f \n", w_back_left_vel);
+    command_interfaces_[2].set_value(w_back_right_vel);
+    fprintf(stderr, " command_interfaces_[2] = %f \n", w_back_right_vel);
+    command_interfaces_[3].set_value(w_front_right_vel);
+    fprintf(stderr, " command_interfaces_[3] = %f \n", w_front_right_vel);
```

Comment:
It is not allowed to print values in the `update` method. Printing is not real-time capable feature.

```suggestion
    command_interfaces_[0].set_value(w_front_left_vel);
    command_interfaces_[1].set_value(w_back_left_vel);
    command_interfaces_[2].set_value(w_back_right_vel);
    command_interfaces_[3].set_value(w_front_right_vel);
```